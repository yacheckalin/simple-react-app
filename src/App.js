import React from "react";
import AuthContainer from "./auth";
import AuthContextProvider from "./auth/auth-context-provider";
import Authorized from "./auth/auth-wrapper";
import TabsContainer from "./tab-container";
import Logout from "./auth/logout";
import Language from "./components/language";

import styled from "styled-components";

const HeaderWrapper = styled.div`
  display: flex;
  box-sizing: border-box;
  justify-content: space-between;
  align-items: center;
  padding: 30px;
`;

const App = () => {
  return (
    <AuthContextProvider>
      <AuthContainer />
      <Authorized>
        <HeaderWrapper>
          <Language />
          <Logout />
        </HeaderWrapper>
        <TabsContainer />
      </Authorized>
    </AuthContextProvider>
  );
};

export default App;
