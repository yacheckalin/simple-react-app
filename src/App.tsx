import React from "react";
import AuthContainer from "./auth";

const App: React.FC = () => (
  <div>
    <AuthContainer />
  </div>
);

export default App;
