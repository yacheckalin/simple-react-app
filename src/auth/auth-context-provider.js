import React, { useState, useContext } from "react";
import { auth } from "./helpers";

const context = {
  isAuthorised: false,
};

export const AuthContext = React.createContext(context);
export const useAuthContext = () => useContext(AuthContext);

const errorResponse = JSON.stringify({
  result: "error",
  error:
    "\u041d\u0435\u0432\u0435\u0440\u043d\u044b\u0435 \u043f\u0430\u0440\u0430\u043c\u0435\u0442\u0440\u044b",
});
const successResponse = JSON.stringify({ result: "ok" });

const AuthContextProvider = ({ children }) => {
  const [isAuthorised, setIsAuthorised] = useState(true);
  const [loginError, setLoginError] = useState("");

  const login = ({ email: login, password }) => {
    //TODO: change for axios request
    let resp = auth({ email: login, password })
      ? successResponse
      : errorResponse;
    const response = Promise.resolve(resp);

    response.then((response) => {
      const res = JSON.parse(response);
      if (res.error) {
        setLoginError(res.error);
        setIsAuthorised(false);
      } else {
        setIsAuthorised(true);
      }
    });
  };

  const logout = ({ email, password }) => {
    //TODO: change for axios request
    const response = Promise.resolve(successResponse);

    response.then((resp) => {
      const res = JSON.parse(resp);

      if (!res.error) {
        setLoginError("");
        setIsAuthorised(false);
      }
    });

    setIsAuthorised(false);
    setLoginError("");
  };

  const resetLoginError = () => {
    setLoginError("");
  };

  return (
    <AuthContext.Provider
      value={{
        isAuthorised,
        setIsAuthorised,
        login,
        logout,
        loginError,
        resetLoginError,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};

export default AuthContextProvider;
