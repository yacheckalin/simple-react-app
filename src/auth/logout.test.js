import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";

import Logout from "./logout";
import App from "../App";
import * as AuthContext from "./auth-context-provider";
import Authorised from "./auth-wrapper";
import AuthForm from "./auth-form";

describe("<Logout />", () => {
  test("displays and renders properly", () => {
    const wrapper = shallow(<Logout />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  test("logout works properly", () => {
    const contextValues = { isAuthorised: true, logout: jest.fn() };
    jest
      .spyOn(AuthContext, "useAuthContext")
      .mockImplementation(() => contextValues);

    const TestElement = () => (
      <div id="test-component">
        <Logout />
      </div>
    );

    const wrapper = mount(
      <>
        <AuthForm />
        <Authorised>
          <TestElement />
        </Authorised>
      </>
    );

    expect(wrapper.find("#test-component")).toBeTruthy();
    wrapper.find("#logout").simulate("click");
    expect(wrapper.find("#email")).toHaveLength(1);
  });
});
