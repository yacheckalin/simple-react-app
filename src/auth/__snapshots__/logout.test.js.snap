// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> displays and renders properly 1`] = `
<styled.div>
  <button
    id="logout"
    onClick={[Function]}
  >
    login.exit
  </button>
</styled.div>
`;
